internal/controller/flux/plugin_controller.go

```diff
@@ -0,0 +1,299 @@
+	pluginController "github.com/cloudoperators/greenhouse/internal/controller/plugin"
+	"github.com/cloudoperators/greenhouse/internal/lifecycle"
+	"github.com/cloudoperators/greenhouse/internal/metrics"
+)
+
+const (
+	defaultNameSpace = "greenhouse"
+	maxHistory       = 10
+	secretKind       = "Secret"
+)
+
+// FluxReconciler reconciles pluginpresets and plugins and translates them into Flux resources
+type FluxReconciler struct {
+	client.Client
+	KubeRuntimeOpts clientutil.RuntimeOptions
+	kubeClientOpts  []clientutil.KubeClientOption
+}
+
+// Greenhouse related RBAC rules for the FluxReconciler
+//+kubebuilder:rbac:groups=greenhouse.sap,resources=plugindefinitions,verbs=get;list;watch
+//+kubebuilder:rbac:groups=greenhouse.sap,resources=plugins,verbs=get;list;watch
+//+kubebuilder:rbac:groups=greenhouse.sap,resources=plugins/status;,verbs=get;update;patch
+//+kubebuilder:rbac:groups=greenhouse.sap,resources=plugins/finalizers,verbs=update
+//+kubebuilder:rbac:groups=greenhouse.sap,resources=clusters;teams,verbs=get;list;watch
+
+// Flux related RBAC rules for the FluxReconciler
+// +kubebuilder:rbac:groups=helm.toolkit.fluxcd.io,resources=helmreleases,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=helm.toolkit.fluxcd.io,resources=helmreleases/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=helm.toolkit.fluxcd.io,resources=helmreleases/finalizers,verbs=get;create;update;patch;delete
+// +kubebuilder:rbac:groups=source.toolkit.fluxcd.io,resources=helmcharts,verbs=get;list;watch
+// +kubebuilder:rbac:groups=source.toolkit.fluxcd.io,resources=helmcharts/status,verbs=get
+// +kubebuilder:rbac:groups=source.toolkit.fluxcd.io,resources=ocirepositories,verbs=get;list;watch
+// +kubebuilder:rbac:groups=source.toolkit.fluxcd.io,resources=ocirepositories/status,verbs=get
+// +kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *FluxReconciler) SetupWithManager(name string, mgr ctrl.Manager) error {
+	r.Client = mgr.GetClient()
+	r.kubeClientOpts = []clientutil.KubeClientOption{
+		clientutil.WithRuntimeOptions(r.KubeRuntimeOpts),
+		clientutil.WithPersistentConfig(),
+	}
+
+	labelSelector := metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      deliveryToolLabel,
+				Operator: metav1.LabelSelectorOpExists,
+			},
+			{
+				Key:      deliveryToolLabel,
+				Operator: metav1.LabelSelectorOpIn,
+				Values:   []string{deliveryToolFlux},
+			},
+		},
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: workqueue.NewTypedMaxOfRateLimiter(
+				workqueue.NewTypedItemExponentialFailureRateLimiter[reconcile.Request](30*time.Second, 1*time.Hour),
+				&workqueue.TypedBucketRateLimiter[reconcile.Request]{Limiter: rate.NewLimiter(rate.Limit(10), 100)}),
+			MaxConcurrentReconciles: 3,
+		}).
```

Comment:
This was mainly to reduce the helm operations done by the helm controller. Therefore we can remove the options here, since Flux is configured outside of Greenhouse.